libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -1,18 +1,37 @@
+
+// tslint:disable-next-line: nx-enforce-module-boundaries
+import { addToReadingList } from './../../../../data-access/src/lib/+state/reading-list.actions';
+// tslint:disable-next-line: nx-enforce-module-boundaries
+import { Book } from './../../../../../shared/models/src/models';
 import { Component } from '@angular/core';
 import { Store } from '@ngrx/store';
 import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 import { ReadingListItem } from '@tmo/shared/models';
+import { MatSnackBar } from '@angular/material/snack-bar';
 @Component({
   selector: 'tmo-reading-list',
   templateUrl: './reading-list.component.html',
   styleUrls: ['./reading-list.component.scss']
 })
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
+  constructor(private readonly store: Store, private snackBar: MatSnackBar) {}
 
-  constructor(private readonly store: Store) {}
-
+  addSnackbar(book) {
+    const snackBarRef = this.snackBar.open(
+      `${book.title} Book has been Removed from the Reading List!`,
+      'Undo',
+      {
+        duration: 3000
+      }
+    )
+    snackBarRef.onAction().subscribe(
+      () =>
+      this.store.dispatch(addToReadingList({ book }))
```

Comment:
For state management cleanliness, try making "undo" its own action instead of repurposing "addToReadingList". Can you also add some unit tests for your NgRx changes?